google/resource_logging_project_sink_test.go

```diff
@@ -0,0 +1,203 @@
+						"google_logging_project_sink.unique_writer"),
+				),
+			},
+		},
+	})
+
+	// Destination and Filter should have changed, but WriterIdentity should be the same
+	if sinkBefore.Destination == sinkAfter.Destination {
+		t.Errorf("Expected Destination to change, but it didn't: Destination = %#v", sinkBefore.Destination)
+	}
+	if sinkBefore.Filter == sinkAfter.Filter {
+		t.Errorf("Expected Filter to change, but it didn't: Filter = %#v", sinkBefore.Filter)
+	}
+	if sinkBefore.WriterIdentity != sinkAfter.WriterIdentity {
+		t.Errorf("Expected WriterIdentity to be the same, but it differs: before = %#v, after = %#v",
+			sinkBefore.WriterIdentity, sinkAfter.WriterIdentity)
+	}
+}
+func testAccCheckLoggingProjectSinkDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_logging_project_sink" {
+			continue
+		}
+
+		attributes := rs.Primary.Attributes
+
+		_, err := config.clientLogging.Projects.Sinks.Get(attributes["id"]).Do()
+		if err == nil {
+			return fmt.Errorf("project sink still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckLoggingProjectSinkExists(n string, sink *logging.LogSink) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		attributes, err := getResourceAttributes(n, s)
+		if err != nil {
+			return err
+		}
+		config := testAccProvider.Meta().(*Config)
+
+		si, err := config.clientLogging.Projects.Sinks.Get(attributes["id"]).Do()
+		if err != nil {
+			return err
+		}
+		*sink = *si
+
+		return nil
+	}
+}
+
+func testAccCheckLoggingProjectSink(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		attributes, err := getResourceAttributes(n, s)
+		if err != nil {
+			return err
+		}
+		config := testAccProvider.Meta().(*Config)
+
+		sink, err := config.clientLogging.Projects.Sinks.Get(attributes["id"]).Do()
+		if err != nil {
+			return err
+		}
+
+		if sink.Destination != attributes["destination"] {
+			return fmt.Errorf("mismatch on destination: api has %s but client has %s", sink.Destination, attributes["destination"])
+		}
+
+		if sink.Filter != attributes["filter"] {
+			return fmt.Errorf("mismatch on filter: api has %s but client has %s", sink.Filter, attributes["filter"])
+		}
+
+		apiLooksUnique := strconv.FormatBool(nonUniqueWriterAccount != attributes["writer_identity"])
+		if apiLooksUnique != attributes["unique_writer_identity"] {
+			return fmt.Errorf("mismatch on unique_writer_identity: api looks like %s but client has %s", apiLooksUnique, attributes["unique_writer_identity"])
+		}
+
+		if sink.WriterIdentity != attributes["writer_identity"] {
+			return fmt.Errorf("mismatch on writer_identity: api has %s but client has %s", sink.WriterIdentity, attributes["writer_identity"])
+		}
+
+		return nil
+	}
+}
+
+func testAccLoggingProjectSink_basic(name, bucketName string) string {
+	return fmt.Sprintf(`resource "google_logging_project_sink" "basic" {
```

Comment:
can you put a line break after the `? (makes it easier to see what's going on and copy and paste too) (here and elsewhere)

Reply:
Done